File: NeetCode/Arrays_and_Hashing/Best_Time_To_Buy_And_Sell_Stocks_II/Best_Time_To_Buy_And_Sell_Stocks_II.cpp
// Best_Time_To_Buy_And_Sell_Stocks_II.cpp
//
// Compile locally with:
// clang++ Best_Time_To_Buy_And_Sell_Stocks_II.cpp -std=c++17 -Wall -Wextra -O2 -o run && ./run
//
// NOTE:
// - Remove main() before submitting to LeetCode.
// - Keep only the class Solution definition.
//

#include <algorithm>
#include <array>
#include <cassert>
#include <cctype>
#include <climits>
#include <cmath>
#include <deque>
#include <functional>
#include <iostream>
#include <map>
#include <numeric>
#include <queue>
#include <set>
#include <stack>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <vector>

using namespace std;

class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int total = 0;
        for (int index = 1; index < prices.size(); index++)
        {
            if (prices[index] > prices[index - 1])
            {
                total += prices[index] - prices[index - 1];
            }
        }
        return total;
    }
};

int main()
{
    cout << "Running Best_Time_To_Buy_And_Sell_Stocks_II.cpp..." << endl;

    Solution sol;

    // TODO:
    // Add local test calls here

    return 0;
}

File: NeetCode/Arrays_and_Hashing/Best_Time_To_Buy_And_Sell_Stocks_II/best__time__to__buy__and__sell__stocks__i_i.py
from typing import Optional, List
import heapq
from collections import defaultdict


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        total = 0
        for index in range(1, len(prices)):
            price = prices[index]
            if price > prices[index - 1]:
                total += price - prices[index - 1]
        return total
        pass


if __name__ == "__main__":
    sol = Solution()
    prices = [1, 2, 3, 4, 5]
    print(sol.maxProfit(prices))
    prices = [7, 1, 5, 3, 6, 4]
    print(sol.maxProfit(prices))
    print("Running Solution...")
